File: pyspider/research/job.py
from dataclasses import dataclass, field
from typing import Any, Dict, List


@dataclass
class ResearchJob:
    seed_urls: List[str]
    site_profile: Dict[str, Any] = field(default_factory=dict)
    extract_schema: Dict[str, Any] = field(default_factory=dict)
    extract_specs: List[Dict[str, Any]] = field(default_factory=list)
    policy: Dict[str, Any] = field(default_factory=dict)
    output: Dict[str, Any] = field(default_factory=dict)

File: gospider/research/job.go
package research

// ResearchJob mirrors the project-oriented research job shape exposed by pyspider.
type ResearchJob struct {
	SeedURLs      []string               `json:"seed_urls"`
	SiteProfile   map[string]interface{} `json:"site_profile,omitempty"`
	ExtractSchema map[string]interface{} `json:"extract_schema,omitempty"`
	ExtractSpecs  []map[string]interface{} `json:"extract_specs,omitempty"`
	Policy        map[string]interface{} `json:"policy,omitempty"`
	Output        map[string]interface{} `json:"output,omitempty"`
}

